compactcert/signer.go

```diff
@@ -178,12 +181,13 @@ func (ccw *Worker) signBlock(hdr bookkeeping.BlockHeader) {
 			continue
 		}
 
-		commitment, err := GenerateStateProofMessage(ccw.ledger, hdr.Round, proto.CompactCertRounds)
+		stateproofMessage, err := GenerateStateProofMessage(ccw.ledger, hdr.Round, proto.CompactCertRounds)
 		if err != nil {
 			ccw.log.Warnf("ccw.signBlock(%d): GenerateStateProofMessage: %v", hdr.Round, err)
 			continue
 		}
-		sig, err := key.StateProofSecrets.SignBytes(commitment)
+		hashedStateproofMessage := stateproofMessage.IntoStateProofMessageHash()
```

Comment:
can we move this + `GenerateStateProofMessage` to be outside to loop? in case we have several accounts, we would reuse the message.